### PR TITLE
Use `@Injectable({providedIn: 'root'})` instead of `TreeModule.forRoot()` to provide `TreeDraggedElement`

### PR DESCRIPTION
**Pull request by KeithGillette**
Since Angular 6, the `@Injectable` decorator has allowed for an optional `providedIn` object parameter  to specify the module in which services should be provided as the as [the preferred alternative](https://angular.io/guide/singleton-services#using-providedin) to the `Providers` array in the `@NgModule()` decorator. This PR provides the `TreeDraggedElement` provider using the new `providedIn` syntax to improve bundle sizes by enabling tree-shaking and allowing Angular Tree Component to bundled in lazy-loaded modules without losing the ability to drag & drop across TreeComponents in different modules that running `TreeDraggedElement` as a Service Singleton allows.

The removal of the `TreeModule.forRoot()` method in this PR introduces a breaking change.

### CODE REVIEW

File: lib/angular-tree-component.ts [hunk 1]
```diff
@@ -66,12 +66,6 @@ import './polyfills';
   providers: []
 })
 export class TreeModule {
-  static forRoot(): ModuleWithProviders {
```
Comment (adamkleingit) [inline lib/angular-tree-component.ts]: Correct me if I'm wrong - this is a breaking change. Anyone using 'forRoot' will have their code break

**Comment by adamkleingit** [comment]
True. Can you please return back the forRoot method for backward compatibility (I don't want to increase a major version just for that)?

Thanks

**Comment by KeithGillette** [comment]
Thanks for responding to this PR so quickly, @adamkleingit. I have restored the `TreeModule.forRoot()` method for backward-compatibility but noted that it's no longer needed in a JSDoc comment.

**Comment by KeithGillette** [comment]
Please let me know if there are any other changes you'd like to see in this PR in order to merge it, @adamkleingit.

**Comment by KeithGillette** [comment]
Hi @adamkleingit — Thanks for your work getting Angular Tree Component working with Ivy and for merging my other PR to enable use of the `DragDropTouch` library with ATC.

Please let me know if you need anything from me to consider merging this PR.

**Comment by KeithGillette** [comment]
Hi @adamkleingit — Just checking-in again on this PR. We've been using a forked version of ATC with this change applied for 3 months without issue.

**Comment by KeithGillette** [comment]
Hi @tobiasengelhardt — I have also rebased this PR on the recent project reorganization. As you can tell from this thread, @adamkleingit did not want to introduce a breaking change by completely removing `TreeModule.forRoot()`. However, if you are anticipating a major version that could introduce breaking changes, I could update this PR to skip the commit that restores that superflous `forRoot()` method. Please let me know your thoughts on this PR.

**Comment by tobiasengelhardt** [comment]
Hi @KeithGillette, we are doing a mayor version soon, so let's completely remove that `TreeModule.forRoot()`

**Comment by KeithGillette** [comment]
Great! I skipped that commit in the update to this PR, so merging will introduce a (trivial to fix) breaking change in the package.

**Pull request merged**